pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/strategy/BalanceNumSegmentAssignmentStrategy.java

```diff
@@ -0,0 +1,82 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix.core.assignment.segment.strategy;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.HelixManager;
+import org.apache.pinot.common.assignment.InstancePartitions;
+import org.apache.pinot.controller.helix.core.assignment.segment.SegmentAssignmentUtils;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.assignment.InstancePartitionsType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Balance num Segment assignment strategy class for offline segment assignment
+ * <ul>
+ *   <li>
+ *     <p>This segment assignment strategy is used when table replication/ num_replica_groups = 1.</p>
+ *   </li>
+ * </ul>
+ */
+public class BalanceNumSegmentAssignmentStrategy implements SegmentAssignmentStrategy {
+  private static final Logger LOGGER = LoggerFactory.getLogger(BalanceNumSegmentAssignmentStrategy.class);
+
+  private String _tableName;
+  private int _replication;
+
+  @Override
+  public void init(HelixManager helixManager, TableConfig tableConfig) {
+    _tableName = tableConfig.getTableName();
+    _replication = tableConfig.getValidationConfig().getReplicationNumber();
+    LOGGER.info("Initialized BalanceNumSegmentAssignmentStrategy for table: "
+            + "{} with replication: {}", _tableName, _replication);
+  }
+
+  @Override
+  public List<String> assignSegment(String segmentName, Map<String, Map<String, String>> currentAssignment,
+      InstancePartitions instancePartitions, InstancePartitionsType instancePartitionsType) {
+    validateSegmentAssignmentStrategy(instancePartitions);
+    return SegmentAssignmentUtils
+        .assignSegmentWithoutReplicaGroup(currentAssignment, instancePartitions, _replication);
+  }
+
+  @Override
+  public Map<String, Map<String, String>> reassignSegments(Map<String, Map<String, String>> currentAssignment,
+      InstancePartitions instancePartitions, InstancePartitionsType instancePartitionsType) {
+    validateSegmentAssignmentStrategy(instancePartitions);
+    Map<String, Map<String, String>> newAssignment;
+    List<String> instances =
+        SegmentAssignmentUtils.getInstancesForNonReplicaGroupBasedAssignment(instancePartitions, _replication);
+    newAssignment = SegmentAssignmentUtils
+        .rebalanceTableWithHelixAutoRebalanceStrategy(currentAssignment, instances, _replication);
+    return newAssignment;
+  }
+
+  private void validateSegmentAssignmentStrategy(InstancePartitions instancePartitions) {
+    int numReplicaGroups = instancePartitions.getNumReplicaGroups();
+    int numPartitions = instancePartitions.getNumPartitions();
+
+    Preconditions.checkState(numReplicaGroups == 1, "Replica groups should be 1");
```

Comment:
Could you add some javadoc to the method on why these two configs must be 1? Or modify the error message here to explain more details like `Replica groups should be 1 in order to use BalanceNumSegmentAssignmentStrategy`?

Reply:
done